core/src/main/java/com/linecorp/armeria/server/http/healthcheck/ManagedHttpHealthCheckService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.healthcheck;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import com.linecorp.armeria.common.http.AggregatedHttpMessage;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * This is more rich implementation of {@link HttpHealthCheckService}.
+ * tl;dr
+ *
+ * <h2>Example as default</h2>
+ * <pre>{@code
+ * Server server = new ServerBuilder()
+ *         .defaultVirtualHost(new VirtualHostBuilder()
+ *                 .serviceUnder("/health", new ManagedHttpHealthCheckService())
+ *                 .build())
+ *         .build();
+ * }</pre>
+ *
+ * <p>Default config uses content on or off
+ * You can also use your own path matching.</p>
+ * <pre>{@code
+ * ...
+ *                 .serviceUnder("health", new ManagedHttpHealthCheckService() {
+ *                         public CompletableFuture&lt;Optional&lt;Boolean&gt;&gt; mode(HttpRequest req) {
+ *                                 return CompletableFuture.completedFuture(Optional.empty());
+ *                         }
+ *                 })
+ *                 .build())
+ * ...
+ * }</pre>
+ */
+public class ManagedHttpHealthCheckService extends HttpHealthCheckService {
+    private static final AggregatedHttpMessage TURN_ON_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn on"));
+    private static final AggregatedHttpMessage TURN_OFF_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn off"));
+
+    @Override
+    protected void doPut(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res) throws Exception {
```

Comment:
How about overriding `doPost` instead? I'm not sure which is a better choice in terms of REST philosophy. What do you think?